docs/src/views/IntegrateCustomComponents.tsx

```diff
@@ -0,0 +1,239 @@
+        '</Provider>',
+      ].join('\n')}
+      render={() => (
+        <Provider
+          theme={{
+            componentVariables: {
+              StyledButton: {
+                color: '#F2F2F2',
+              },
+            },
+            componentStyles: {
+              StyledButton: {
+                root: ({ props, variables, theme: { siteVariables } }) => ({
+                  backgroundColor: siteVariables.brand,
+                  color: (variables as any).color,
+                }),
+              },
+            },
+          }}
+        >
+          <StyledButton>Provider styled button</StyledButton>
+        </Provider>
+      )}
+    />
+
+    <p>
+      The second argument of the <code>createComponent</code> config param is the{' '}
+      <code>render</code> method. This is the place where where you might link Stardust bits with
+      your custom component - e.g. by simply passing them as props. This <code>render</code> method
+      will be invoked with the following parameters:
+    </p>
+    <ul>
+      <li>
+        <code>stardust</code> - the object containing the evaluated theming props (<code>
+          classes
+        </code>
+        and <code>rtl</code>).
+      </li>
+      <li>
+        <code>...props</code> - all other props provided by the user.
+      </li>
+    </ul>
+
+    <Header as="h2" content="Using the custom components" />
+
+    <p>
+      We already saw how the <code>Provider</code> can define some stylings and variables for the
+      custom components. Next, we will take a look into several examples of how the user can further
+      customize styles and variables of these components, the same way they would do with the
+      Stardust components.
+    </p>
+
+    <Header
+      as="h3"
+      content={
+        <>
+          Example 1. Using <code>styles</code> property
+        </>
+      }
+    />
+
+    <ExampleSnippet
+      value={[
+        `<StyledButton styles={{backgroundColor: "black"}}>`,
+        `  Inline styled button`,
+        `</StyledButton>`,
+      ].join('\n')}
+      render={() => (
+        <Provider
+          theme={{
+            componentVariables: {
+              StyledButton: {
+                color: '#F2F2F2',
+              },
+            },
+            componentStyles: {
+              StyledButton: {
+                root: ({ props, variables, theme: { siteVariables } }) => ({
+                  backgroundColor: siteVariables.brand,
+                  color: (variables as any).color,
+                }),
+              },
+            },
+          }}
+        >
+          <StyledButton styles={{ backgroundColor: 'black' }}>Inline styled button</StyledButton>
+        </Provider>
+      )}
+    />
+
+    <Header
+      as="h3"
+      content={
+        <>
+          Example 2. Using <code>variables</code> property
+        </>
+      }
+    />
+
+    <ExampleSnippet
+      value={[
+        `<StyledButton variables={{color: "black" }}>`,
```

Comment:
just to be sure - could you, please, suggest where is the code that suggests to the user where this variable is used (i.e. styles code)?

Reply:
Added paragraph with Provider and example componentStyles that would be applied.